llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -522,7 +529,8 @@ bool lowerEsimdConstructs(module_split::ModuleDesc &MD) {
 IrPropSymFilenameTriple saveModule(module_split::ModuleDesc &MD, int I,
                                    StringRef IRFilename = "") {
   IrPropSymFilenameTriple Res;
-  StringRef Suffix = MD.isESIMD() ? "_esimd" : "";
+  StringRef Suffix =
+      MD.isDoubleGRF() ? "_esimd_x2grf" : (MD.isESIMD() ? "_esimd" : "");
```

Comment:
Should the first check be MD.isDoubleGRF() && MD.isESIMD()

Reply:
isDoubleGRF() can return true only when it is ESIMD is also true. That field is set in getESIMDDoubleGRFSplitter()
LowerESIMD.cpp:L49 adds ATTR_DOUBLE_GRF only for ESIMD functions